src/Symfony/Component/Messenger/Transport/KernelTerminate/SymfonyKernelEventTransportFactory.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\KernelTerminate;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Messenger\Transport\MemoryTransport;
+use Symfony\Component\Messenger\Transport\TransportFactoryInterface;
+use Symfony\Component\Messenger\Transport\TransportInterface;
+
+class SymfonyKernelEventTransportFactory implements TransportFactoryInterface
```

Comment:
```diff
- SymfonyKernelEventTransportFactory
+ SymfonyTerminateTransportFactory
```

IMHO, unless we decide to make it more configurable. But then it could be any event(s) :)
As already explained before, I think this is the only use-case that really fits in core and needs to be provided out-of-the-box by the framework config.